test/integration/src/test/java/io/pravega/test/integration/ReaderGroupNotificationTest.java

```diff
@@ -188,19 +206,93 @@ public void testSegmentNotifications() throws Exception {
         assertEquals(2, segmentNotificationPostScale.getNumOfSegments());
     }
 
+    @Test(timeout = 40000)
+    public void testTargetRateNotifications() throws Exception {
+        final String streamName = "stream1";
+        Random random = new Random(0);
+        StreamConfiguration config = StreamConfiguration.builder()
+                .scalingPolicy(ScalingPolicy.byEventRate(1, 2, 3))
+                .build();
+        Controller controller = controllerWrapper.getController();
+        controllerWrapper.getControllerService().createScope(SCOPE).get();
+        controller.createStream(SCOPE, streamName, config).get();
+        @Cleanup
+        ConnectionFactory connectionFactory = new SocketConnectionFactoryImpl(ClientConfig.builder()
+                .controllerURI(URI.create("tcp://localhost"))
+                .build());
+        @Cleanup
+        ClientFactoryImpl clientFactory = new ClientFactoryImpl(SCOPE, controller, connectionFactory);
+        EventStreamWriter<String> writer1 = clientFactory.createEventWriter(streamName, new UTF8StringSerializer(),
+                EventWriterConfig.builder().build());
+        List<String> routingKeys = new ArrayList<>();
+        routingKeys.add("0");
+        routingKeys.add("1");
+        routingKeys.add("2");
+        routingKeys.add("3");
+        writer1.writeEvent(routingKeys.get(0), "data").get();
+        writer1.writeEvent("0", "data").get();
+        long start = System.currentTimeMillis();
+        log.info("Segments before scale found as: {}", controller.getCurrentSegments(SCOPE, streamName).get());
+        while (System.currentTimeMillis() - start < Duration.ofSeconds(20).toMillis()) {
+            writer1.writeEvent(routingKeys.get(random.nextInt(3)), "data").get();
+        }
+
+        // scale
+        List<List<Segment>> segmentsList = scale(controller, SCOPE, streamName, 1, 1, executorService());
```

Comment:
The scale method here performs a single manual scale operation by invoking   `controller.scaleStream(...)` and does not rely on auto-scaling by the controller. 
{ Even though the stream is configured with an auto-scaling policy(`ScalingPolicy.byEventRate(1, 2, 3)`) }
Hence this test is similar to the already existing test `io.pravega.test.integration.ReaderGroupNotificationTest#testSegmentNotifications`.

If the intent is to verify notifications with auto-scaling of the stream by the controller then I think it would be better if it is implemented as a system test. We should not invoke the manual scale API of the controller for such a test.